lib/relation-definition.js

```diff
@@ -2401,30 +2409,37 @@ RelationDefinition.embedsMany = function embedsMany(modelFrom, modelTo, params)
 
   // validate all embedded items
   if (definition.options.validate !== false) {
-    modelFrom.validate(propertyName, function(err) {
+    modelFrom.validateAsync(propertyName, function(err, done) {
       var self = this;
       var embeddedList = this[propertyName] || [];
       var hasErrors = false;
-      embeddedList.forEach(function(item, idx) {
+
+      async.forEachOf(embeddedList, function(item, idx, eachCb) {
         if (item instanceof modelTo) {
-          if (!item.isValid()) {
-            hasErrors = true;
-            var id = item[idName];
-            var first = Object.keys(item.errors)[0];
-            if (id) {
-              var msg = 'contains invalid item: `' + id + '`';
-            } else {
-              var msg = 'contains invalid item at index `' + idx + '`';
+          item.isValid(function(valid) {
+            if (!valid) {
+              hasErrors = true;
+              var id = item[idName];
+              var first = Object.keys(item.errors)[0];
+              if (id) {
+                var msg = 'contains invalid item: `' + id + '`';
+              } else {
+                var msg = 'contains invalid item at index `' + idx + '`';
+              }
+              msg += ' (`' + first + '` ' + item.errors[first] + ')';
+              self.errors.add(propertyName, msg, 'invalid');
             }
-            msg += ' (`' + first + '` ' + item.errors[first] + ')';
-            self.errors.add(propertyName, msg, 'invalid');
-          }
+            eachCb();
```

Comment:
No this is correct as this is adding the msg to the errors array on the model instance if it was added it would fail after the first validation error and not run them all.